BLSQ/iaso
# POLIO-1586 add lqas statuses

Explain what problem this PR is resolving

Related JIRA tickets : POLIO-1586

## Self proofreading checklist


## Doc
- Tell us where the doc can be found (docs folder, wiki, in the code...)

## Changes

- LQAS country view
    - change key for failed status from `"3lqasfail"`  to  '"3lqasverypoor"`, following API change
    - handle new `3lqasoversampled` and `3lsasundersampled`statuses:
        - In list view: show status, with same color code as no data
        - In map view: show over- and undersampled distrcts in hashed grey
        - In map view: Add "No valid data" color code in map legend  
    - Related graphs have not been changed since changing the way data is computed should be changd on OpenHexa side 
    - Add components to enable using multiple patterns as shape backgrounds:
        - `PaneWithPatterns`: a leaflet `Pane` wrapped to be able to use patterns with SVG
        - `HashedPattern`: Component to make hashed patterns with configurable angle, stroke width and colors
        - `HashedGreyPattern`: grey hashed pattern ready to be used with `PaneWithPatterns`
    - Replaced `LQAS_FAIL` with  `LQAS_VERY_POOR` in country map. Didn't change in afro map, since it only has passed or failed status 
     

## How to test

 - Replace you local polio DB datastore for org unit 29709 with [https://staging.poliooutbreaks.com/api/datastore/lqas_29709/](https://staging.poliooutbreaks.com/api/datastore/lqas_29709/)
 - Go to LQAS >country and select the right country
 - Check round 1: list view should show over and undersampled districts, map view should show them with a hashed pattern
 
## Print screen / video

![Screenshot 2024-06-26 at 16 13 20](https://github.com/BLSQ/iaso/assets/38907762/fa2e1511-ad49-47df-a2f1-1b7ea07136c1)
